docs/feature-flags.md

```diff
@@ -0,0 +1,116 @@
+# 🚦 Feature Flagging — Tech Docs
+
+This document explains how our **client‑side** feature‐flag system works, how flags are stored, and how to add or consume a flag in daily development.
+
+---
+
+## 1. What lives where?
+
+```
+src/
+  feature-flags/
+    flags.ts     # canonical list, default states, metadata
+    store.ts     # runtime state resolver (localStorage + URL)
+    panel.tsx    # React component to disable and enable a feature flag
+    hooks.ts     # React hooks and component to disable code
+```
+
+*There is **no** backend call. Everything happens in the browser.*
+
+---
+
+## 2. Persistence & update flow
+
+| Step | Source           | Example                              | Notes                                        |
+| ---- | ---------------- | ------------------------------------ | -------------------------------------------- |
+| 1    | **Default**      | `defaultEnabled` in `flags.ts`       | Baseline value shipped in the bundle.        |
+| 2    | **localStorage** | `__ff_overrides__ => { "foo": true }` | Persists your choices across reloads.        |
+| 3    | **URL params**   | `?ff=foo,bar` or `?ff_foo=false`    | Highest priority; great for shareable links. |
+
+**Setting a flag with our API does two things:**
+
+1. Writes to localStorage so the change sticks.
+2. Patches the current URL (`history.replaceState`) so the address bar shows the new state without a full reload.
+
+---
+
+### URL grammar cheat‑sheet
+
+* `?ff=flagA,flagB` → turn *on* multiple flags quickly.
+* `?ff_flagA=true|false` → explicit on/off for one flag.
```

Comment:
whether the flagA/flagB here is the key of feature flag, like kubearchive and buildServiceAccount? 

It seems we could enable feature on any URL of UI? 
I tried the  https://localhost:8080/ns?ff=buildServiceAccount locally but nothing happens. 
What is the right url i could see the panel?

Reply:
yes, you can enable features from any of the routes using the use Params. To see the panel click the flask icon in the toolbar.

Reply:
> I tried the https://localhost:8080/ns?ff=buildServiceAccount locally but nothing happens.

Do we have any features on the namespace page related to build service account

Reply:
https://localhost:8080/ns/wlin-tenant/secrets/create?ff=buildServiceAccount  does not work for me, either.